src/test/java/com/uwetrottmann/tmdb2/BaseTestCase.java

```diff
@@ -1,123 +1,117 @@
 package com.uwetrottmann.tmdb2;
 
-import com.uwetrottmann.tmdb2.entities.CastMember;
-import com.uwetrottmann.tmdb2.entities.CrewMember;
-import com.uwetrottmann.tmdb2.entities.Image;
-import com.uwetrottmann.tmdb2.entities.Media;
-import com.uwetrottmann.tmdb2.entities.Videos;
+import com.google.common.util.concurrent.RateLimiter;
+import okhttp3.Interceptor;
 import okhttp3.OkHttpClient;
 import okhttp3.logging.HttpLoggingInterceptor;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.text.ParseException;
+import java.util.Vector;
+
+import static com.uwetrottmann.tmdb2.TmdbTestSuite.authenticatedInstance;
+import static com.uwetrottmann.tmdb2.TmdbTestSuite.initialized;
+import static com.uwetrottmann.tmdb2.TmdbTestSuite.suiteRunning;
+import static com.uwetrottmann.tmdb2.TmdbTestSuite.unauthenticatedInstance;
+import static com.uwetrottmann.tmdb2.TmdbTestSuite.DEBUG;
 
-import java.util.List;
+public abstract class BaseTestCase {
 
-import static org.assertj.core.api.Assertions.assertThat;
+    protected final Tmdb getUnauthenticatedInstance() {
+        return unauthenticatedInstance;
+    }
 
-public abstract class BaseTestCase {
+    protected final Tmdb getAuthenticatedInstance() {
+        return authenticatedInstance;
+    }
+
+    private static int testClassToRun = 0;
+
+    @BeforeClass
+    public static void ensureEnvironmentIsSet() throws IOException, ParseException {
+        if (suiteRunning)
+            return;
+
+        if (initialized)
+            return;
+
+        try {
+            Field field = ClassLoader.class.getDeclaredField("classes");
+            field.setAccessible(true);
+
+            System.out.println("Tests Running Standalone.");
 
-    // Do NOT use this API key in your application, it is solely for testing tmdb-java!
-    private static final String API_KEY = "25da90e9f8f0b3892d8bdeb6c3d6267d";
+            System.out.println("Test Classes To Run:");
 
-    private static final boolean DEBUG = true;
+            @SuppressWarnings({ "unchecked", "rawtypes" })
+            Vector<Class> classes = (Vector<Class>) field.get(BaseTestCase.class.getClassLoader());
+            for (Class<?> clazz : classes) {
+                if (!clazz.getSimpleName().equals("Test") && clazz.getName().endsWith("Test")) {
+                    System.out.println("\t"+clazz.getSimpleName());
+                    System.out.println("\t\tTests:");
+                    for (Method method : clazz.getDeclaredMethods()) {
+                        if (method.isAnnotationPresent(Test.class)) {
+                            System.out.println("\t\t\t"+method.getName());
```

Comment:
Generally to run the whole package, it requires a full environment initialization, so when not running a suite, you need to determine after how many classes you have to destruct the environment.  Having said that, prints are not necessary and will be removed. They were some sort of debugging for me.